processing/src/main/java/io/druid/query/extraction/LookupExtractor.java

```diff
@@ -25,16 +25,18 @@
 
 import javax.annotation.Nullable;
 import javax.validation.constraints.NotNull;
+import java.io.Closeable;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = LookupDelegator.class)
 @JsonSubTypes(value = {
-    @JsonSubTypes.Type(name = "map", value = MapLookupExtractor.class)
+    @JsonSubTypes.Type(name = "map", value = MapLookupExtractor.class),
+    @JsonSubTypes.Type(name = "registered", value = LookupDelegator.class)
 })
-public abstract class LookupExtractor
+public abstract class LookupExtractor implements Closeable
```

Comment:
There is no guarantee that the manager will not close a lookup extractor after it has been fetched. What is the intended outcome if the manager closes the LookupExtractor before it can be used?